### DISCUSSION TITLE
Pattern of URL tokens

### DISCUSSION THREAD

**Discussion by OlesYudin** [thinkst/canarytokens#590] [Q&A] [answered]
Hi, how do you generate a path for a URL token type of decoy? I found some patterns during my test, when you generate a URL token you will see the next URL pattern:
```
- https://canarytoken.com/static/about/images/<some_id>/index.html
- https://canarytoken.com/terms/<some_id>/submit.aspx
- https://canarytokens.com/feedback/static/<some_id>/post.jsp
- https://canarytokens.com/stuff/terms/<some_id>/contact.php
- https://canarytokens.com/articles/about/<some_id>/index.html
- https://canarytokens.com/tags/static/terms/<some_id>/post.jsp
- https://canarytokens.com/images/<some_id>/payments.js
```
What other pattern do you have? I want to create a redirection rule from my domain to canarytokens based on all URL patterns and by default redirect to my static home page in case the URL pattern doesn't meet my rule.

**Comment by roberto-aldera** [comment, marked as answer]
Hey @OlesYudin - the random URL generation is implemented [here](https://github.com/thinkst/canarytokens/blob/c191914a301f9771a3936e2e8ef2daac1bca9da3/canarytokens/canarydrop.py#L341-L371), where path elements are read from the DB.
